proto/task_execution.proto

```diff
@@ -23,6 +23,11 @@ message TaskParameter {
   //data is a directory, if used for an output all the files in the directory
   //will be copied to the storage location
   bool   directory = 5;
+  //if the parameter is an output, should the element be created before executing
```

Comment:
Sorry, but just trying to clarify for myself, what is the difference between a `location`, a `path` and a `directory` and how should they be used?  And where are `data` and volumes defined?

```Proto
  //location in long term storage, is a url specific to the implementing 
  //system. For example s3://my-object-store/file1 or gs://my-bucket/file2 or
  //file:///path/to/my/file
  string location = 3;
  //path in the machine file system. Note, this MUST be a path that exists
  //within one of the defined volumes
  string path = 4;
  //data is a directory, if used for an output all the files in the directory
  //will be copied to the storage location
  bool directory = 5;
```

Thanks,
~p